File: Paragraph.cpp
#include "Paragraph.hpp"

#include "AlignLeft.hpp"
#include "AlignRight.hpp"

Paragraph::Paragraph(std::string text)
{
  this->text = text;
}

void Paragraph::add(const Element&)
{
  throw "Not Allowed!";
}

void Paragraph::remove(const Element&)
{
  throw "Not Allowed!";
}

Element& Paragraph::get(size_t)
{
  throw "Not Allowed!";
}

void Paragraph::set_alignment(const AlignStrategy& alignment)
{
  if (dynamic_cast<AlignLeft*>((AlignStrategy*)&alignment)) this->alignment = "LeftAlignment";
  else if (dynamic_cast<AlignRight*>((AlignStrategy*)&alignment)) this->alignment = "RightAlignment";
}

void Paragraph::write(std::ostream& out)
{
  out << "Paragraph: " << this->text << " Alignment: " << this->alignment;
}

File: main_project/models/Paragraph.java
package main_project.models;

import main_project.models.interfaces.Element;

public class Paragraph implements Element {
    private final String text;

    public Paragraph(String text) {
        this.text = text;
    }

    @Override
    public void print() {
        System.out.println("Paragraph: " + text);
    }

    @Override
    public void add(Element e) {
        throw new RuntimeException("Paragraph.add not allowed");
    }

    @Override
    public void remove(Element e) {
        throw new RuntimeException("Paragraph.remove not allowed");
    }

    @Override
    public Element get(int idx) {
        throw new RuntimeException("Paragraph.get not allowed");
    }
}
